ping/main.go

```diff
@@ -0,0 +1,284 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/libp2p/go-libp2p"
+	"github.com/libp2p/go-libp2p/p2p/protocol/ping"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+
+	"github.com/libp2p/go-libp2p-noise"
+	"github.com/libp2p/go-libp2p-secio"
+	tls "github.com/libp2p/go-libp2p-tls"
+
+	"github.com/testground/sdk-go/network"
+	"github.com/testground/sdk-go/run"
+	"github.com/testground/sdk-go/runtime"
+	"github.com/testground/sdk-go/sync"
+)
+
+var testcases = map[string]interface{}{
+	"ping": run.InitializedTestCaseFn(runPing), // we don't need the type conversion, but it's here for instructional purposes.
+}
+
+// A test plan is just a standard Go program which receives a bunch of
+// environment variables to configure itself. A test plan ONLY depends on the
+// lightweight Testground SDK (https://github.com/testground/sdk-go/).
+//
+// A central object is the RunEnv (runtime environment), which encapsulates the
+// contract between Testground and this test plan. Read more: https://docs.testground.ai/concepts-and-architecture/runtime.
+//
+// Other key objects are:
+//
+//  * sync.Client: used to coordinate instances with one another.
+//  * network.Client: used to manipulate network configurations.
+func main() {
+	// Delegate this run to the SDK. InvokeMap takes a map of test case names
+	// and test case functions, and dispatches accordingly depending on the test
+	// case being run.
+	run.InvokeMap(testcases)
+}
+
+// runPing is the test case logic.
+//
+// Its signature conforms to the SDK's run.InitializedTestCaseFn type. As a
+// result, the Testground SDK will perform a few useful preparation steps
+// for us:
+//
+//  1. Initializing a sync client, bound to the runenv.
+//  2. Initializing a net client.
```

Comment:
Can we clarify a bit more on the difference between a sync and a net client? (They kind of sound very similar to discern in the reader's mental model, I'm probably missing some background here though.)

Reply:
Was explained in the `main()` function godocs, but it doesn't hurt to expand here. Will do.

<img width="568" alt="image" src="https://user-images.githubusercontent.com/1101242/88197169-ff7ff700-cc39-11ea-835d-9c76427a4ce1.png">